shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -583,86 +695,118 @@ def start(self):
         if move_line:
             message = self.msg_store.recovered_previous_session()
             return self._response_for_set_quantity(move_line, message=message)
-        return self._response_for_select_location()
+        return self._response_for_select_location_or_package()
 
-    def scan_location(self, barcode):
-        """Scan a source location.
+    def scan_location_or_package(self, barcode):
+        """Scan a source location or a source package.
 
         It is the starting point of this scenario.
 
-        If stock has been found in the scanned location, allows to scan a
-        product or a lot.
+        If stock has been found in the scanned location, or if a package has been found,
+        it allows to scan a product or a lot.
 
         Transitions:
         * select_product: to scan a product or a lot stored in the scanned location
         * start: no stock found or wrong barcode
         """
         search = self._actions_for("search")
+        package = search.package_from_scan(barcode)
+        if package:
+            return self._scan_package(package)
         location = search.location_from_scan(barcode)
-        handlers = [
-            self._scan_location__location_found,
-            self._scan_location__check_location,
-            self._scan_location__check_stock,
-        ]
-        response = self._use_handlers(handlers, location)
-        if response:
-            return response
-        return self._response_for_select_product(location)
+        quants_in_location = self.env["stock.quant"].search(
+            [("location_id", "=", location.id), ("quantity", ">", 0)]
+        )
+        return self._scan_location(location, quants_in_location)
```

Comment:
This method `scan_location_or_package` should also use search.find and return barcode not found as fallback